snuba/snuba_migrations/replays/0015_index_frequently_accessed_columns.py

```diff
@@ -0,0 +1,40 @@
+from typing import Iterator, Sequence
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return list(forward_columns_iter())
+
+    def backwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return list(backward_columns_iter())
+
+
+def forward_columns_iter() -> Iterator[operations.SqlOperation]:
+    for column in materialized_columns:
+        yield operations.AddIndex(
+            storage_set=StorageSetKey.REPLAYS,
+            table_name="replays_local",
+            index_name=f"bf_{column}",
+            index_expression=column,
+            index_type="bloom_filter()",
+            granularity=1,
+            target=operations.OperationTarget.LOCAL,
+        )
+
+
+def backward_columns_iter() -> Iterator[operations.SqlOperation]:
+    for column in materialized_columns:
+        yield operations.DropIndex(
+            StorageSetKey.REPLAYS,
+            "replays_local",
+            f"bf_{column}",
+            target=operations.OperationTarget.LOCAL,
+        )
+
+
+materialized_columns = ["urls", "user_id", "user_name", "user_email", "ip_address_v4"]
```

Comment:
It seems excessive to have indexes on all these columns. For example, I'd imaging user_name and user_email are tied to the same user_id. Do we need indexes on user_name and user_email? Adding indexes incurs a cost on ingest and queries that don't use them.

Reply:
I could remove urls and user_name.  I think urls wouldn't benefit too much and user_name is not regularly searched.  user id/ip/email are three of our most poplar search fields so they should stay.